### PR TITLE
storage: allow linear reverse seek in lock cf

### PR DESCRIPTION
**Pull request by BusyJay**
When seeking lock cf reversely, it's still a linear seek, so we can optimize it using the same trick as #1126 does.

@disksing @zhangjinpeng1987 PTAL

### CODE REVIEW

**Comment by BusyJay** [comment]
PTAL

**Comment by zhangjinpeng1987** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @disksing

File: src/storage/mvcc/reader.rs [hunk 1]
```diff
@@ -49,7 +49,7 @@ impl<'a> MvccReader<'a> {
             return Ok(vec![]);
         }
         if self.scan_mode.is_some() && self.data_cursor.is_none() {
-            self.data_cursor = Some(try!(self.snapshot.iter(None, self.scan_mode.unwrap())));
+            self.data_cursor = Some(try!(self.snapshot.iter(None, self.get_scan_mode(false))));
```
Comment (disksing) [inline src/storage/mvcc/reader.rs]: Why not turn it on for default cf?

**Comment by disksing** [comment]
LGTM

**Pull request merged**